src/apiErrors.ts

```diff
@@ -0,0 +1,11 @@
+import { Action } from './step'
+
+export type ToolErrorType = 'NO_QUOTE'
+
+export interface ToolError {
+  errorType: ToolErrorType
+  code: string
```

Comment:
Should we type the code or would it be too annoying to add new ones?
Advantage would be that the dev knows what to expect

Reply:
Yeah the reason I made it a string it to avoid the whole breaking changes thing when we wanna update something. We need docs though if we don't expose the type.